phac-nml/irida-next
# Member expiration frontend

## What does this PR do and why?
Fixes #284.
Added date picker input to the members table for each member & create member modal.

## Screenshots or screen recordings

Groups
![Screenshot from 2023-12-20 14-22-45](https://github.com/phac-nml/irida-next/assets/325703/4d0e35a2-0cd0-4877-813a-088e1ece547f)
![Screenshot from 2023-12-20 14-24-44](https://github.com/phac-nml/irida-next/assets/325703/4bdc9eee-7f5f-4870-b2db-3b0ea2eb11c2)

Projects
![Screenshot from 2023-12-20 14-28-02](https://github.com/phac-nml/irida-next/assets/325703/45cb0b38-3180-4ca7-9b72-7ae47be840a7)
![Screenshot from 2023-12-20 14-30-29](https://github.com/phac-nml/irida-next/assets/325703/c756012b-df1d-448a-96c7-3aa2541a9320)

## How to set up and validate locally
Modify a project/group member:
1. Navigate to a manageable project or group.
2. Click `members` from the navigation menu.
3. Modify the expiration date of a projects/group member.
4. Verify the member was successfully updated.

Create a new project/group member:
1. Click the `Add New Member` button.
2. Fill in the form with & without the expiration date.
3. Verify the member was successfully created.

*Note: Try back dating the expiration date and signing in as the expired user. Verify the expired user does not have access to the project/group.*

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
